components/script/dom/htmltableelement.rs

```diff
@@ -398,16 +398,20 @@ impl HTMLTableElementMethods for HTMLTableElement {
     // https://html.spec.whatwg.org/multipage/#dom-table-deleterow
     fn DeleteRow(&self, mut index: i32) -> Fallible<()> {
         let rows = self.Rows();
+
         // Step 1.
-        if index == -1 {
-            index = rows.Length() as i32 - 1;
-        }
-        // Step 2.
-        if index < 0 || index as u32 >= rows.Length() {
+        if index < -1 || index >= rows.Length() as i32 {
```

Comment:
This could potentially use a range to make things more idiomatic:

```suggestion
        let num_rows = rows.Length() as i32;
        if !(-1..num_rows).contains(index) {
```